verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -42,25 +51,177 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
+        "ERROR: generate-variants only works on one file.");
+  }
+  const char* source_file = args[0];
+  std::string source_contents;
+  if (auto status = verible::file::GetContents(source_file, &source_contents);
+      !status.ok()) {
+    std::cerr << "ERROR: passed file can't be open\n.";
+    return status;  // check if the the file is not readable or doesn't exist.
+  }
+
+  // Lexing the input SV source code.
+  verilog::VerilogLexer lexer(source_contents);
+  verible::TokenSequence lexed_sequence;
+  for (lexer.DoNextToken(); !lexer.GetLastToken().isEOF();
+       lexer.DoNextToken()) {
+    // For now we will store the syntax tree tokens only, ignoring all the
+    // white-space characters. however that should be stored to output the
+    // source code just like it was.
+    if (verilog::VerilogLexer::KeepSyntaxTreeTokens(lexer.GetLastToken())) {
+      lexed_sequence.push_back(lexer.GetLastToken());
+    }
+  }
+
+  // Control flow tree constructing.
+  verilog::FlowTree control_flow_tree(lexed_sequence);
+  int counter = 0;
+  auto status = control_flow_tree.GenerateVariants(
+      [limit_variants, &counter](const verilog::FlowTree::Variant& variant) {
+        if (counter == limit_variants) return false;
+        counter++;
+        std::cout << "Variant number " << counter << ":\n";
+        for (auto token : variant.sequence) std::cout << token << '\n';
+        puts("");
+        return true;
+      });
+  if (!status.ok()) {
+    return status;
+  }
 
-Input:
+  return absl::OkStatus();
+}
+
+static const std::pair<absl::string_view, SubcommandEntry> kCommands[] = {
+    {"strip-comments",
+     {&StripComments,
+      R"(strip-comments file [replacement-char]
+Inputs:
   'file' is a Verilog or SystemVerilog source file.
   Use '-' to read from stdin.
-
+  'replacement-char' is a character to replace comments with.
+  If not given, or given as a single space character, the comment contents and
+  delimiters are replaced with spaces.
+  If an empty string, the comment contents and delimiters are deleted. Newlines
+  are not deleted.
+  If a single character, the comment contents are replaced with the character.
 Output: (stdout)
   Contents of original file with // and /**/ comments removed.
 )"}},
+    {"multiple-cu",
+     {&MultipleCU,
+      R"(multiple-cu file [more_files]
+Inputs:
+  'file' is a Verilog or SystemVerilog source file.
+   There can be multiple SystemVerilog source files.
+   Each one of them will be compiled in a separate compilation unit.
+Output: (stdout)
+  The preprocessed files content (same contents with directives interpreted).
+)"}},
+    {"generate-variants",
+     {&GenerateVariants,
+      R"(genera-variants file [-limit_variants number]
+Inputs:
+  'file' is a Verilog or SystemVerilog source file.
+  '-limit_variants' flag limits variants to 'number' (20 by default).
+Output: (stdout)
+   Generates every possible variant considering the conditional directives.
+)"}},
```

Comment:
We have an opportunity for another sub-command: `list-macros`, which would be the output of the `GetUsedMacros()` function from the FlowTree (can be a separate change)